File: python/level2/2_biggestnum.py
#가장 큰 수
#https://programmers.co.kr/learn/courses/30/lessons/42746


import itertools


def solution(numbers):
    num_str = [str(num) for num in numbers]
    # print(num_str[0]*4)
    num_str.sort(key=lambda x: (x * 4)[:4], reverse=True)
    # print(num_str)

    if num_str[0] != '0':
        answer = "".join(num_str)
        return answer
    else:
        return "0"


    return answer

File: java/level2/2_biggestnum.java
//가장 큰 수
//https://programmers.co.kr/learn/courses/30/lessons/42746?language=java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        ArrayList<String> arr = new ArrayList<>();
        for (int i : numbers) {
            arr.add(i+"");
        }
        Collections.sort(arr, new Comparator<String>() {
            public int compare(String o1, String o2) {
                return (o2+o2+o2+o2).substring(0,4).compareTo((o1+o1+o1+o1).substring(0,4));
            }
        });
        if(arr.get(0).equals("0")) return "0";
        for (String i : arr) {
            answer += i;
        }
        
        
        return answer;
    }
}